### ISSUE TITLE
Optuna pruning callback

### ISSUE THREAD

**Issue by ThomasMeissnerDS** [dreamquark-ai/tabnet#318]
## Feature request

**What is the expected behavior?**
Currently I optimize TabNet with Optuna. However it takes lots of time. Having a pruning callback for Optuna would help a lot here.

**What is motivation or use case for adding/changing the behavior?**
Reduced training times and better performance.

**How should this be implemented in your opinion?**
I am not sure, if it's possible to set this up already, but similar to custom loss function.

**Are you willing to work on this yourself?**
No, unless I can implement it from what is in the library already (consumer side)

**Comment by Optimox** [comment]
Hello,

Could you explain a bit further what you mean by « a pruning callback »?

**Comment by thomasmeissnercrm** [comment]
Optuna offers an API where a trained model can give feedback about the current loss during training. If Optuna estimates, that this will not lead to a better trial overall, it will abort the trial and try and new set of hyperparameters.This saves lots of time as many hyperparameter sets will not be tried fully, but end rather quickly.

For Pytorch there is the optuna.integration.PyTorchLightningPruningCallback i.e.

For reference:
https://optuna.readthedocs.io/en/stable/reference/integration.html

**Comment by Optimox** [comment]
Ok I see.

There is a `callbacks` parameter in the fit function. I don't think the callback from pytorch ligthning can be used as is but you can for sure adapt it to pytorch-tabnet. You'll find the main class for callbacks with some example (as early stopping callback) here : https://github.com/dreamquark-ai/tabnet/blob/develop/pytorch_tabnet/callbacks.py

If you need help do not hesitate to ask.
